### PR TITLE
DeleteEnvironmentApplicationLock Role Based Access Control

### PR DESCRIPTION
**Pull request by bernardo-bastos**
Add RBAC to the DeleteEnvironmentApplicationLock action.

Rev: DSN-J1WO8X

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -763,12 +763,18 @@ func (c *CreateEnvironmentApplicationLock) Transform(ctx context.Context, state
...
 
 type DeleteEnvironmentApplicationLock struct {
+	Authentication
 	Environment string
 	Application string
 	LockId      string
 }
 
 func (c *DeleteEnvironmentApplicationLock) Transform(ctx context.Context, state *State) (string, error) {
+	// TODO (BB): add the "*" check on the checkUserPermissions() method.
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:773]: handle todo

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: services/cd-service/pkg/repository/transformer_test.go [hunk 2]
```diff
@@ -794,30 +794,30 @@ func TestRbacTransformerTest(t *testing.T) {
...
 				&CreateEnvironmentLock{
-					Environment:    "production",
-					Message:        "don't",
-					LockId:         "manual",
-					Authentication: Authentication{RBACConfig: auth.RBACConfig{DexEnabled: true, Policy: map[string]*auth.Permission{}}},
+					Environment: "production",
+					Message:     "don't",
+					LockId:      "manual",
+					Authentication: Authentication{RBACConfig: auth.RBACConfig{DexEnabled: true, Policy: map[string]*auth.Permission{
+						"developer,CreateLock,production:production,*,allow": {Role: "developer"}}}},
```
[the remark this answers is not in the document]
Comment (bernardo-bastos) [reply services/cd-service/pkg/repository/transformer_test.go]: Was just to keep the order consistent. I removed the change.

File: services/cd-service/pkg/repository/transformer.go [hunk 3]
```diff
@@ -763,12 +763,18 @@ func (c *CreateEnvironmentApplicationLock) Transform(ctx context.Context, state
...
+	err := state.checkUserPermissions(ctx, c.Environment, "*", auth.PermissionDeleteLock, c.RBACConfig)
```
[the remark this answers is not in the document]
Comment (bernardo-bastos) [reply services/cd-service/pkg/repository/transformer.go]: Added the fix.

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (bernardo-bastos) [reply services/cd-service/pkg/repository/transformer.go]: Already handled and removed the comment.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**